### ISSUE TITLE
[SmarterMail100] Smartermail SM100 Provider unable to import a domain that SM10 Provider can

### ISSUE THREAD

**Issue by enguard1** [FuseCP/SolidCP#205]
**Describe the bug**
When using the SM10 Provider, we are able to import a domain from the Smartermail server. When we switch to the latest SM100 Provider, it does not import the domain and ends with an error message in Event Viewer. I have enabled verbose output in the web.config file on the Smartermail server. 

**To Reproduce**
1. Using the SM Provider and successfully importing the domain, here is the verbose output in the Event Viewer:
[12/4/2024 3:59:46 PM] START: 'SmarterMail 10.x +' Install
[12/4/2024 3:59:46 PM] END: 'SmarterMail 10.x +' Install
[12/4/2024 3:59:46 PM] START: GetMemory
[12/4/2024 3:59:46 PM] END: GetMemory
[12/4/2024 4:00:15 PM] START: 'SmarterMail 10.x +' GetDomains
[12/4/2024 4:00:15 PM] END: 'SmarterMail 10.x +' GetDomains
[12/4/2024 4:00:20 PM] START: 'SmarterMail 10.x +' GetAccounts
[12/4/2024 4:00:20 PM] END: 'SmarterMail 10.x +' GetAccounts
[12/4/2024 4:00:20 PM] START: 'SmarterMail 10.x +' GetGroups
[12/4/2024 4:00:20 PM] END: 'SmarterMail 10.x +' GetGroups
[12/4/2024 4:00:20 PM] START: 'SmarterMail 10.x +' GetLists
[12/4/2024 4:00:20 PM] END: 'SmarterMail 10.x +' GetLists
[12/4/2024 4:00:20 PM] START: 'SmarterMail 10.x +' GetDomainAliases
[12/4/2024 4:00:20 PM] END: 'SmarterMail 10.x +' GetDomainAliases
[12/4/2024 4:00:20 PM] START: 'SmarterMail 10.x +' GetAccounts
[12/4/2024 4:00:20 PM] END: 'SmarterMail 10.x +' GetAccounts
[12/4/2024 4:00:22 PM] START: 'SmarterMail 10.x +' GetMailAliases
[12/4/2024 4:00:22 PM] END: 'SmarterMail 10.x +' GetMailAliases
[12/4/2024 4:00:22 PM] START: 'SmarterMail 10.x +' GetGroups
[12/4/2024 4:00:22 PM] END: 'SmarterMail 10.x +' GetGroups
[12/4/2024 4:00:22 PM] START: 'SmarterMail 10.x +' GetLists
[12/4/2024 4:00:22 PM] END: 'SmarterMail 10.x +' GetLists

2. Using the SM100 Provider and not able to import the same domain, here is the verbose output in the Event Viewer:
[12/4/2024 4:05:09 PM] START: 'SmarterMail 100.x +' Install
[12/4/2024 4:05:09 PM] END: 'SmarterMail 100.x +' Install
[12/4/2024 4:05:09 PM] START: GetMemory
[12/4/2024 4:05:09 PM] END: GetMemory
[12/4/2024 4:05:29 PM] START: 'SmarterMail 100.x +' GetDomains
[12/4/2024 4:05:29 PM] END: 'SmarterMail 100.x +' GetDomains
[12/4/2024 4:05:33 PM] START: 'SmarterMail 100.x +' GetAccounts
[12/4/2024 4:05:33 PM] INFO: GetAccounts - Account: user@domain.com (disguised) 

 DomainAdmin: False 

 ImportDomainAdmin: False
***IT DOES THIS FOR THE NEXT 400 USERS***********
[12/4/2024 4:05:33 PM] END: 'SmarterMail 100.x +' GetAccounts
[12/4/2024 4:05:33 PM] START: 'SmarterMail 100.x +' GetGroups
[12/4/2024 4:05:33 PM] ERROR: 'SmarterMail 100.x +' GetGroups
System.Exception: Could not get the list of mail domain groups ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at SolidCP.Providers.Mail.SmarterMail100.GetGroups(String domainName)
   --- End of inner exception stack trace ---
   at SolidCP.Providers.Mail.SmarterMail100.GetGroups(String domainName)
   at SolidCP.Server.MailServer.GetGroups(String domainName)

**Expected behavior**
The SM100 Provider should be able to import any domain from a smartermail server. 

**Screenshots**
<img width="1053" alt="Screenshot 2024-12-04 at 4 12 23 PM" src="https://github.com/user-attachments/assets/229c2ff4-ad87-403c-bbd2-8318d6e941a0">

**SolidCP Info**
 - SolidCP Version:  1.5.0

**Desktop (please complete the following information):**
 - OS: Windows Server 2022
 - Browser: Edge
 - Version: 131.0.2903.70

**Additional context**
Temporarily switching to the SM10 Provider works and we can import the domain.

**Comment by trevorrobinson** [comment]
Hey,

Just for future reference normally proving the error reported in eventviewer is enough to save copy and pasting the full sequence but sometimes the extra details can help.

I tried to recreate this but wasnt able to in our dev environment. If you try import a smaller domain does it error also?

Regards,

Trevor

**Comment by FuseCP-TRobinson** [comment]
Hey,

Just for future reference normally proving the error reported in eventviewer is enough to save copy and pasting the full sequence but sometimes the extra details can help.

I tried to recreate this but wasnt able to in our dev environment. If you try import a smaller domain does it error also?

Regards,

Trevor

**Comment by enguard1** [comment]
I found the problem! When an Alias on Smartermail is empty, the SM100 Provider cannot import the domain. See attached screeshots.

![Screenshot 2024-12-04 at 5 51 32 PM](https://github.com/user-attachments/assets/4da4c9d0-522b-4dcb-bfcb-049a6ac29dcf)
![Screenshot 2024-12-04 at 5 51 23 PM](https://github.com/user-attachments/assets/cf294ffa-81a9-41ba-80fd-c83f14b2f675)

I'm guessing the SM10 Provider ignores those empty aliases and imports anyway.

I deleted the empty aliases on the Smartermail server and was able to import the domain with the SM100 Provider!

There are 2 scenarios where we need to detach and import:

1. Customer changes their domain name entirely
2. Customer changes a user address name. (e.g. bob@example.com to john@example.com)
